elastic/kibana
# [Security Solution]]Detection Engine] adds telemetry for alert suppression

## Summary

- addresses https://github.com/elastic/security-team/issues/9201
- mappings for snapshot telemetry https://github.com/elastic/telemetry/pull/3800
- adds snapshot telemetry for alert suppression
- adds EBT (Event Based Telemetry),
 telemetry event sent on rule execution when suppression is enabled and alerts created/suppresssed

### Testing

1. Setup rules with suppression
2. Run rules, ensure alerts created and suppressed
3. Check telemetry:
  - Snapshot
      For snapshot telemetry use [API](https://docs.elastic.dev/telemetry/collection/snapshot-telemetry#telemetry-usage-payload-api) call
      **OR**
      Check snapshots in Kibana adv settings -> Global Settings  Tab -> Usage collection section -> Click on **cluster data** example link -> Check suppression fields in snapshot in flyout
      
  - EBT
      Check for event `event_type : "alert_suppression_on_rule_execution"` in `ebt-kibana-server` index or `security-solution-ebt-kibana-server` data view on telemetry staging host.
      [Link](https://telemetry-v2-staging.elastic.dev/s/securitysolution/app/discover#/?_g=(time:(from:now-7d,to:now))&_a=(columns:!(),filters:!(),index:security-solution-ebt-kibana-server,interval:auto,query:(language:kuery,query:'event_type%20:%20%22alert_suppression_on_rule_execution%22%20%20'),sort:!(!(timestamp,desc)))) to data view in Discover for telemetry staging

#### EBT event

  ```json
  {
    "suppressionAlertsCreated": 4,
    "suppressionAlertsSuppressed": 20,
    "suppressionRuleName": "Custom rule",
    "suppressionDuration": 1200,
    "suppressionFieldsNumber": 1,
    "suppressionFields": ["host.name"],
    "suppressionRuleType": "esql",
    "suppressionMissingFields": true,
    "suppressionMRuleId": "5d4b1f60-6b3e-4a97-ab89-161a13b6fafa"
  }
  ```

- suppressionDuration is stored in seconds. For configured suppression per rule execution it is stored as `-1`
- suppressionRuleName for custom rules will be recorded as `Custom rule` and full name for prebuilt rules


### Staging dashboard

- https://telemetry-v2-staging.elastic.dev/s/securitysolution/app/dashboards#/view/bafef410-4fb0-4625-8fb1-cf81368f20c8?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-2w,to:now))
